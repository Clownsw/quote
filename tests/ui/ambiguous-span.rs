use proc_macro2::Span;
use quote::quote_spanned;

trait CallSite {
    fn get() -> Self;
}

impl CallSite for Span {
    fn get() -> Self {
        Span::call_site()
    }
}

fn main() {
    let _ = quote_spanned!(CallSite::get()=> ...);
}
