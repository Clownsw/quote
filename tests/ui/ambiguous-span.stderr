error[E0283]: type annotations needed
  --> tests/ui/ambiguous-span.rs:15:28
   |
15 |     let _ = quote_spanned!(CallSite::get()=> ...);
   |             ---------------^^^^^^^^^^^^^^^-------
   |             |              |
   |             |              cannot infer type
   |             required by a bound introduced by this call
   |
   = note: cannot satisfy `_: IntoSpan`
   = help: the following types implement trait `IntoSpan`:
             DelimSpan
             Span
